build: pin Rust toolchain / libsqlite3-sys to avoid cfg_select! breakage on stable < 1.96

## Problem

Loom's installer compiles `loom-daemon` from source as part of the Quick Install / `loom-daemon init` step. This build pulls in `libsqlite3-sys 0.38.0` (via the workspace's transitive deps), whose `build.rs` uses the `cfg_select!` library macro.

`cfg_select!` is **unstable in stable Rust versions below approximately 1.96**. On Rust 1.92.0 (Dec 2025), the build fails with:

```
error[E0658]: use of unstable library feature `cfg_select`
   --> /Users/rwalters/.cargo/registry/src/.../libsqlite3-sys-0.38.0/build.rs:110:9
    |
110 |         cfg_select! {
    |         ^^^^^^^^^^
    |
    = note: see issue #115585 <https://github.com/rust-lang/rust/issues/115585> for more information
```

The full failure is a hard install blocker: `loom-daemon` won't build → installer reports `Installation failed at step: Initialize Loom` → the install rolls back. The CI on the loom main branch passes because CI uses a recent enough Rust; only local operators on an older stable Rust hit it silently.

## Reproduction (this session)

During the v0.9.1 sphere install (https://github.com/rjwalters/loom/releases/tag/v0.9.1):

- Local Rust: `rustc 1.92.0 (ded5c06cf 2025-12-08)`, `cargo 1.92.0 (344c4567c 2025-10-21)`.
- Install command: `./scripts/install-loom.sh --force --allow-active-session --allow-stale-target -y ../sphere`
- Failure: at the "Initialize Loom" step (`Building loom-daemon...`), cargo failed to compile `libsqlite3-sys 0.38.0`.
- Workaround: `rustup update stable` (bumped to `rustc 1.96.0 (ac68faa20 2026-05-25)`). The next install attempt succeeded.

This means **every operator on stable < 1.96 will hit this silently** until they `rustup update`.

## Proposed fix

Pick one of:

**Option 1 (preferred): add `rust-toolchain.toml` pinning to a known-good minimum.**

```toml
# rust-toolchain.toml at repo root
[toolchain]
channel = "1.96.0"  # or "1.96" / "stable" — whichever matches CI's actual toolchain
```

This forces every contributor and every operator's build to use a version that supports `cfg_select!`. `rustup` will auto-install the required toolchain when first invoked in the repo.

**Option 2: pin `libsqlite3-sys` to a pre-0.38 version in `Cargo.toml`** (or in the workspace `Cargo.toml` if it's a workspace dep). This keeps the toolchain lax but downgrades the offending dep to a version that doesn't use `cfg_select!`.

Option 2 is less invasive (no toolchain change for contributors) but Option 1 is more robust — it'll catch the next dep that adopts a recently-stabilized feature too.

**Either way**: add an MSRV (Minimum Supported Rust Version) note to the README / CONTRIBUTING.md so contributors and operators know what's required.

## Acceptance Criteria

- [ ] `./scripts/install-loom.sh` succeeds against a fresh target on the current minimum supported Rust.
- [ ] CI matches the chosen MSRV (either matches the `rust-toolchain.toml`, or has an explicit MSRV check).
- [ ] CONTRIBUTING.md / README documents the MSRV.
- [ ] If Option 1: `rust-toolchain.toml` lives at the repo root and is committed.
- [ ] If Option 2: `Cargo.toml` (or workspace `Cargo.toml`) explicitly pins `libsqlite3-sys` to a known-good range.

## Out of Scope

- Migrating to `rusqlite` (the high-level wrapper) or switching SQLite bindings entirely. Out of scope here.
- Cleaning up the chain of dependencies that brought in `libsqlite3-sys 0.38.0` — that's an inventory question for a separate issue if the dep is unnecessary.

## Related

- Encountered during the Phase 4 sphere install (#3382 — just closed).

## Affected Files

- **New file**: `rust-toolchain.toml` at repo root — add `[toolchain]\nchannel = "1.96.0"`
- `.github/workflows/ci.yml` — currently uses `dtolnay/rust-toolchain@stable` (lines 51, 66); no change needed if `rust-toolchain.toml` is present (the action respects the file), but verify it still works
- `README.md` or `CONTRIBUTING.md` — add MSRV note ("Requires Rust 1.96.0 or newer")

## Implementation Decision

**Go with Option 1** (rust-toolchain.toml). Rationale:
- `rusqlite 0.40` → `libsqlite3-sys 0.38.0` is the culprit; the dep can't be downgraded without dropping `rusqlite` minor version
- A toolchain file is the canonical Rust solution, `rustup` auto-installs the required toolchain on first use
- Pinning to `1.96.0` (released 2026-05-25) matches the current CI environment where builds pass

## Risk

Low. The toolchain file is a developer ergonomic change; production builds are unaffected. CI already passes on stable ≥ 1.96.

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -53,7 +53,9 @@ Run `pnpm check:ci` before pushing. This runs the full CI suite locally (linting
 
 If you want to explore the codebase or run Loom locally:
 
-**Prerequisites**: Node.js 20+, pnpm, Rust (stable), tmux, Git, GitHub CLI (`gh`)
+**Prerequisites**: Node.js 20+, pnpm, Rust 1.96.0 or newer (see `rust-toolchain.toml`), tmux, Git, GitHub CLI (`gh`)
+
+> **Minimum Supported Rust Version (MSRV)**: Rust 1.96.0. The `loom-daemon` crate depends on `rusqlite` → `libsqlite3-sys`, which uses `cfg_select!` (stabilised in Rust 1.96). Building with an older stable toolchain will fail. Run `rustup update stable` or `rustup install 1.96.0` if needed. The `rust-toolchain.toml` at the repo root pins the toolchain automatically when you use `rustup`.
 
 ```bash
 git clone https://github.com/rjwalters/loom.git

--- a/rust-toolchain.toml
+++ b/rust-toolchain.toml
@@ -1,0 +1,2 @@
+[toolchain]
+channel = "1.96.0"

